### ISSUE TITLE
Optional chaining function call with generic isn't handled properly

### ISSUE THREAD

**Issue by br0p0p** [swc-project/swc#4083]
### Describe the bug

> There's a similar issue that could be relevant here: #2734

There seems to be a gap in syntax handling for the combination of optional function calls and generics. The issue occurs when trying to call a potentially undefined generic function and passing a generic type variable.

See the code snippet for a full example but the compiler seems to crash when it hits the following string of characters: 

`?.<`

as in 

```ts
foo.get?.<number>()
```

I imagine this syntax is very rarely used but it is valid Typescript.

Here's the Typescript Playground equivalent: https://www.typescriptlang.org/play?#code/DYUwLgBAZg9jBcEDeEDm4D8iA8AVAfABQAe8uAlBALz4S4QC+1yDA3AFADGMAdgM6Q+MALbgAFgEseqZmABOAVxAd2EqBEJDRYSdMpJ2EI9DgA6dJCrQFPTmAm8IeIqTrlE9A8eNzwCuTwQxBzeDOxh7LAw5pim2DwKwgBGIHJEAEzkQA

### Input code

```typescript
let foo: { get?: <T>(x:T) => T } = {};
const something = true;

if (something) {
    foo.get = function <T>(x: T): T {
      return x;
    }
}

foo.get?.<number>(2);
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false
    },
    "target": "es2022",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.2.106&code=H4sIAAAAAAAAAz2MQQrEIBRD9%2F8UWdZNF7PU1l7CE0z5doSpgv1CQbx7ZZAJZJGQvC8LfEoaFQfLprE4O93aKawWDg0rajO0p3gJrnSyfEI8eiu5sCEKHtO%2FVqiErg6cO62vfIm7hBQHFk51jxWQWUqOuM0vN2pE47rNSyznm7OdXuoBc33z66QAAAA%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1FsWLLnKHHsK4SnDxD40CDcZ3rx1cspNmnp4qfeBorVSsgGVMOJPaL62kZ2E48UVpIUWPNhvArS9WdtaOMMzDmF6HnMETWCj65LdzyFyORRi4K5v28Cdbd8X8PkZQr3%2BX80ntdsw7j9cEVQ5uPx%2Bqcl%2B4AAAA

### Expected behavior

This syntax is properly compiled to an optional function call (for any valid target) stripping out types as appropriate.

### Actual behavior

The compiler fails to compile this syntax and throws an error.

### Version

1.2.106

**Comment by magic-akari** [comment]
It has been fixed since [1.2.144](https://play.swc.rs/?version=1.2.144&code=H4sIAAAAAAAAAz2MQQrEIBRD9%2F8UWdZNF7PU1l7CE0z5doSpgv1CQbx7ZZAJZJGQvC8LfEoaFQfLprE4O93aKawWDg0rajO0p3gJrnSyfEI8eiu5sCEKHtO%2FVqiErg6cO62vfIm7hBQHFk51jxWQWUqOuM0vN2pE47rNSyznm7OdXuoBc33z66QAAAA%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1FsWLLnKHHsK4SnDxD40CDcZ3rx1cspNmnp4qfeBorVSsgGVMOJPaL62kZ2E48UVpIUWPNhvArS9WdtaOMMzDmF6HnMETWCj65LdzyFyORRi4K5v28Cdbd8X8PkZQr3%2BX80ntdsw7j9cEVQ5uPx%2Bqcl%2B4AAAA).

**Comment by kdy1** [comment]
@magic-akari Thank you!
